server/src/integration/java/com/networknt/server/ConsulRegistryIT.java

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright (c) 2016 Network New Technologies Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.networknt.server;
+
+import com.networknt.cluster.LightCluster;
+import com.networknt.config.Config;
+import com.networknt.consul.ConsulConfig;
+import com.networknt.consul.ConsulRegistry;
+import com.networknt.consul.client.ConsulClientImpl;
+import com.networknt.registry.URL;
+import com.networknt.registry.URLImpl;
+import com.networknt.utility.Constants;
+import com.networknt.utility.Util;
+import org.junit.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.GenericContainer;
+
+import java.net.InetAddress;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import static com.networknt.consul.ConsulConstants.CONFIG_NAME;
+import static com.networknt.server.Server.STATUS_HOST_IP;
+import static org.junit.Assert.fail;
+
+/**
+ * This test is an automatic test for real consul registry and discovery by using testContainers.
+ * Since it depends on docker so it should be disabled all the time unless it is used.
+ */
+@Ignore
```

Comment:
Is there a better way to enable/disable then (what I presume) is manually removing the @ Ignore flag? What is the process for running full integration tests?

Reply:
We can enable the test case. If you don't want to execute the integration test cases, use `mvn clean package`. If you want to run all integration tests along with unit tests, then you can use `mvn clean install`.

Reply:
Maybe we don't want to go that far in case someone wants to install the dependencies but doesn't have docker installed.. could we add a maven profile for running integration tests?

Reply:
Using maven profile is another option. The key point is we need to ensure that broader users can compile without the docker installed and our developers can trigger integration tests with a switch. I really have no idea which commands most developers using `mvn package` or `mvn install`.

Reply:
@stevehu Hey. I think `mvn install` would be more popular to use. So I updated the pom to disable the integration test execution when using `mvn install`.  

In this way,`mvn install` only run the unit test.
And the following commands are using for executing integration tests:
mvn verify -DskipITs=false
mvn integration-tests -DskipITs=false
mvn clean install -DskipITs=false

What do you think?

Reply:
I think it is a very good idea. mvn verify -DskipITs=false can execute integration tests when they are annotated as ignored?

Reply:
I just tested. If `@ignore` annotated inside testing code. There seems no way to execute them. I think we can remove all the ignore annotation in the integration test if the default value for skipITs is true.

Reply:
Got it. I like the idea. We need to make sure that everything is transparent to new developers.